internal/db/content.go

```diff
@@ -13,22 +13,22 @@ import (
 )
 
 type Hash struct {
-	H1 []byte
-	H2 []byte
+	H1 [16]byte
+	H2 [16]byte
 }
 
 func HashContent(data []byte) Hash {
 	sha := sha256.Sum256(data)
 	return Hash{
-		H1: sha[0:16],
-		H2: sha[16:32],
+		H1: *(*[16]byte)(sha[0:16]),
+		H2: *(*[16]byte)(sha[16:32]),
```

Comment:
This hack is only possible since Go 1.17 https://www.jetbrains.com/go/guide/tips/go-1-17-convert-slice-to-array-pointer/?playlist=new-features-in-goland-2021-2

But it avoids copying these values into a new array.